//*****************************************************************************
// Copyright 2025 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//*****************************************************************************

syntax = "proto2";
package mediapipe;

import "mediapipe/framework/calculator.proto";


message ImageGenCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    // https://github.com/google/mediapipe/issues/634 have to be unique in app
    // no rule to obtain this
    optional ImageGenCalculatorOptions ext = 116423751;
    }

    // fields required for GenAI pipeline initialization
    required string models_path = 1;
    optional string device = 2;
    optional string plugin_config = 3;

    // fields used during inference
    optional string max_resolution = 4 [default = "4096x4096"];
    optional string default_resolution = 5;
    optional uint64 max_num_images_per_prompt = 6 [default = 10];
    optional uint64 default_num_inference_steps = 7 [default = 50];
    optional uint64 max_num_inference_steps = 8 [default = 100];

    // static reshape setting, required for NPU, optional for other devices
    optional string resolution = 9;
    optional int64 num_images_per_prompt = 10;
    optional float guidance_scale = 11;
}
